### PR TITLE
use shellcheck JSON output

### PR DESCRIPTION
**Pull request by euclio**

### CODE REVIEW

**Comment by blueyed** [comment]
Thanks!
https://github.com/euclio/neomake/blob/329d77437e2a9c2c56facb8028d1af4f5028b00c/tests/integration.vader#L183 fails, but it is also a not so good test by itself.  I will look into fixing it (removing shellcheck there) later.

**Comment by blueyed** [comment]
https://github.com/neomake/neomake/pull/2304 (merged).
Please rebase.

**Comment by euclio** [comment]
Rebased.

File: autoload/neomake/makers/ft/sh.vim [hunk 1]
```diff
@@ -40,6 +36,30 @@ function! neomake#makers#ft#sh#shellcheck() abort
...
+            let type = 'W'
+        else
+            let type = 'E'
+        endif
+
+        call add(errors, {
+            \ 'text': err.message,
+            \ 'lnum': err.line,
+            \ 'col': err.column,
+            \ 'length': err.endColumn - err.column,
```
Comment (blueyed) [inline autoload/neomake/makers/ft/sh.vim]: This needs `+1`.

File: autoload/neomake/makers/ft/sh.vim [hunk 2]
```diff
@@ -40,6 +36,30 @@ function! neomake#makers#ft#sh#shellcheck() abort
...
+            \ 'filename': err.file,
```
Comment (blueyed) [inline autoload/neomake/makers/ft/sh.vim]: The `filename` part does not work with temporary files.
Likely setting/using `bufnr` instead would work / be better.

**Comment by blueyed** [comment]
Given the above issues: what is the benefit of using the json output?  Only having the length easily?

**Comment by euclio** [comment]
Yeah, being able to calculate the length is the main reason.

**Comment by blueyed** [comment]
Re `endColumn` support in general: https://github.com/koalaman/shellcheck/issues/966

**Comment by blueyed** [comment]
And regarding tab size: https://github.com/koalaman/shellcheck/issues/1048

**Comment by blueyed** [comment]
Left a comment at https://github.com/koalaman/shellcheck/issues/1048#issuecomment-461550298.

File: autoload/neomake/makers/ft/sh.vim (hunk 2 above)
Comment (blueyed) [reply autoload/neomake/makers/ft/sh.vim]: This should be fixed via https://github.com/neomake/neomake/pull/2305.

**Comment by euclio** [comment]
Since endColumn support is on its way out, I think the downsides of this outweigh the benefits. Closing.

**Comment by blueyed** [comment]
Agreed.

> Since endColumn support is on its way out

Do you have a source?

**Comment by euclio** [comment]
https://github.com/koalaman/shellcheck/issues/966, but I didn't realize that it was from over a year ago. I guess it's no longer accurate.

**Pull request closed without merging**